src/processor/template_processor/base/base_template_processor.py

```diff
@@ -284,6 +284,10 @@ def populate_all_template_snapshot(self):
             root_dir_path = self.repopath 
 
         self.paths = get_field_value(self.node, 'paths')
+        self.type = get_field_value(self.node, 'type')
+        
+        if self.type in ["yaml","json"] and self.paths  is None:
```

Comment:
The validation is already there for check `self.paths` is exist or not.
Just update the error message in Line no: 302

Reply:
condition removed and the type will check in another func